draft/ZEP0002.md

```diff
@@ -0,0 +1,285 @@
+    that is divisble by `chunks_per_shard` per dimension, e.g. starting at the zero-origin.
+    The length of the array must match the length of the array metadata `shape` entry.
+    For example, a value `[32, 2]` indicates that 64 chunks are combined in one shard,
+    32 along the first dimension, and for each of those 2 along the second dimension.
+    Valid starting positions for a shard in the chunk-grid are therefore `[0, 0]`,
+    `[32, 2]`, `[32, 4]`, `[64, 2]` or `[96, 18]`.
+
+
+### Binary shard format
+
+The following part is cited from the accompanying sharding extension specification [[SHARDINGSPEC](#ref-SHARDINGSPEC)]:
+
+> The only binary format is the ``indexed`` format, as specified by the ``type``
+configuration key. Other binary formats might be added in future versions.
+
+> In the indexed binary format chunks are written successively in a shard, where
+unused space between them is allowed, followed by an index referencing them.
+The index is placed at the end of the file and has a size of 16 bytes multiplied by the number of chunks
+in a shard, for example ``16 bytes * 64 = 1014 bytes`` for ``chunks_per_shard=[32, 2]``.
+The index holds an `offset, nbytes` pair of little-endian uint64 per chunk,
+the chunks-order in the index is row-major (C) order, for example for
+``chunks_per_shard=[2, 2]`` an index would look like:
+
+> ```
+| chunk (0, 0)    | chunk (0, 1)    | chunk (1, 0)    | chunk (1, 1)    |
+| offset | nbytes | offset | nbytes | offset | nbytes | offset | nbytes |
+| uint64 | uint64 | uint64 | uint64 | uint64 | uint64 | uint64 | uint64 |
+```
+
+> Empty chunks are denoted by setting both offset and nbytes to ``2^64 - 1``.
+The index always has the full shape of all possible chunks per shard,
+even if they are outside of the array size.
+
+> The actual order of the chunk-content is not fixed and may be chosen by the implementation
+as all possible write orders are valid according to this specification and therefore can
+be read by any other implementation. When writing partial chunks into an existing shard no
+specific order of the existing chunks may be expected. Some writing strategies might be
+
+> * **Fixed order**: Specify a fixed order (e.g. row-, column-major or Morton order).
+    When replacing existing chunks larger or equal sized chunks may be replaced in-place,
+    leaving unused space up to an upper limit  which might possibly be specified.
+    Please note that for regular-sized uncompressed data all chunks have the same size and
+    can therefore be replaced in-place.
+> * **Append-only**: Any chunk to write is appended to the existing shard,
+    followed by an updated index.
+
+> Any configuration parameters for the write strategy must not be part of the metadata document,
+they need to be configured at runtime, as this is implementation specific.
+
+
+### Use-cases and examples
+
+
+#### Reading data per chunk
```

Comment:
not sure whether this subsection fits the "use cases and examples" section :thinking:

Reply:
True, I simply renamed the section to `Use-cases and access strategies`